google-cloud-clients/google-cloud-contrib/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/it/ITGcsNio.java

```diff
@@ -538,6 +539,52 @@ public void testListFiles() throws IOException {
     }
   }
 
+  @Test
+  public void testRelativityOfResolve() throws IOException {
+    try (FileSystem fs = getTestBucket()) {
+      Path abs1 = fs.getPath("/dir");
+      Path abs2 = abs1.resolve("subdir/");
+      Path rel1 = fs.getPath("dir");
+      Path rel2 = rel1.resolve("subdir/");
+      // children of absolute paths are absolute,
+      // children of relative paths are relative.
+      assertThat(abs1.isAbsolute()).isTrue();
+      assertThat(abs2.isAbsolute()).isTrue();
+      assertThat(rel1.isAbsolute()).isFalse();
+      assertThat(rel2.isAbsolute()).isFalse();
+    }
+  }
+
+  @Test
+  public void testWalkFiles() throws IOException {
+    try (FileSystem fs = getTestBucket()) {
+      List<Path> goodPaths = new ArrayList<>();
+      List<Path> paths = new ArrayList<>();
+      goodPaths.add(fs.getPath("dir/angel"));
+      goodPaths.add(fs.getPath("dir/alone"));
+      paths.add(fs.getPath("dir/dir2/another_angel"));
+      paths.add(fs.getPath("atroot"));
+      paths.addAll(goodPaths);
+      goodPaths.add(fs.getPath("dir/dir2/"));
+      for (Path path : paths) {
+        fillFile(storage, BUCKET, path.toString(), SML_SIZE);
+      }
+      // Given a relative path as starting point, walkFileTree must return only relative paths.
+      List<Path> relativePaths = postTraversalWalker.walkFileTree(fs.getPath("dir/"));
+      for (Path p : relativePaths) {
+        assertWithMessage("Should have been relative: " + p.toString()).that(p.isAbsolute()).isFalse();
+      }
+      assertThat(relativePaths.size()).isEqualTo(5);
```

Comment:
Do you think it's worth explicitly listing out the paths?

Reply:
I think it's preferable to keep this test focused on only one thing. There'a already another test that checks that walkFiles finds all the files.